src/providers/twitch/PubsubClient.cpp

```diff
@@ -1428,23 +1431,45 @@ void PubSub::handleMessageResponse(const rapidjson::Value &outerData)
                 rapidjson::Value senderData;
                 if (!rj::getSafeObject(messageData, "sender", senderData))
                 {
-                    qDebug() << "Failed to get sender";
+                    qCDebug(chatterinoPubsub) << "Failed to get sender";
                     return;
                 }
-                QString sender_id;
-                if (!rj::getSafe(senderData, "user_id", sender_id))
+                QString senderId;
+                if (!rj::getSafe(senderData, "user_id", senderId))
                 {
-                    qDebug() << "Failed to get sender user id";
+                    qCDebug(chatterinoPubsub) << "Failed to get sender user id";
                     return;
                 }
-                QString sender_login;
-                if (!rj::getSafe(senderData, "login", sender_login))
+                QString senderLogin;
+                if (!rj::getSafe(senderData, "login", senderLogin))
                 {
-                    qDebug() << "Failed to get sender login";
+                    qCDebug(chatterinoPubsub) << "Failed to get sender login";
                     return;
                 }
-                action.target = ActionUser{sender_id, sender_login};
-                qDebug() << action.msgID;
+                QString senderDisplayName;
+                if (!rj::getSafe(senderData, "display_name", senderDisplayName))
+                {
+                    qCDebug(chatterinoPubsub)
+                        << "Failed to get sender display name";
+                    return;
+                }
+                QString senderColor;
+                if (!rj::getSafe(senderData, "chat_color", senderColor))
+                {
+                    qCDebug(chatterinoPubsub)
+                        << "Failed to get sender color, might be a grey-name";
+                    // return; // color might be empty if user is a grey-name, don't freak out
+                }
+                // handling non-ascii usernames
+                if (QString::compare(senderLogin, senderDisplayName,
+                                     Qt::CaseInsensitive) != 0)
+                {
+                    senderDisplayName =
+                        QString("%1(%2)").arg(senderLogin, senderDisplayName);
```

Comment:
This should respect the username style setting

Reply:
I believe I sorted this out in bb004cc
![proof](https://cdn.zneix.eu/0WnX3Ja.png)